lib/rspec/core/bisect/runner.rb

```diff
@@ -67,17 +74,22 @@ def run_locations(*capture_args)
           end
         end
 
+        # Consider just using Process.spawn since Open3 uses that.
         # `Open3.capture2e` does not work on JRuby:
         # https://github.com/jruby/jruby/issues/2766
         if Open3.respond_to?(:capture2e) && !RSpec::Support::Ruby.jruby?
           def run_command(cmd)
+            # consider passing in { 'SPEC_OPTS' => spec_opts } when not spec_opts.empty?
+            # instead of mutating SPEC_OPTS
             Open3.capture2e(cmd).first
           end
-        else # for 1.8.7
+        else # for 1.8.7 and JRuby
           # :nocov:
           def run_command(cmd)
             out = err = nil
 
+            # consider passing in { 'SPEC_OPTS' => spec_opts } when not spec_opts.empty?
+            # instead of mutating SPEC_OPTS
```

Comment:
Didn't work first time and didn't think till right now that I can test effect of setting any env vars for arbitrary command, not just adding the hash and watching tests fail....

B mobile phone

> On Jun 4, 2016, at 12:46 PM, Myron Marston <notifications@github.com> wrote:
> 
> In lib/rspec/core/bisect/runner.rb:
> 
> >            # :nocov:
> >            def run_command(cmd)
> >              out = err = nil
> >  
> > +            # consider passing in { 'SPEC_OPTS' => spec_opts } when not spec_opts.empty?
> > +            # instead of mutating SPEC_OPTS
> Any reason this can't be done now?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>